draft-ietf-quic-recovery.md

```diff
@@ -483,11 +483,15 @@ the peer spurious timeouts.  The maximum ack delay is communicated in the
 An acknowledgement SHOULD be sent immediately upon receipt of a second
 packet but the delay SHOULD NOT exceed the maximum ack delay. QUIC recovery
 algorithms do not assume the peer generates an acknowledgement immediately when
-receiving a second full-packet.
+receiving a second packet.
 
 Out-of-order packets SHOULD be acknowledged more quickly, in order to accelerate
 loss recovery.  The receiver SHOULD send an immediate ACK when it receives a new
-packet which is not one greater than the largest received packet number.
+packet which is not one greater than the largest received packet number.  The
+receiver MAY send an immediate ACK when a missing packet begins exceeding the
+peer's reordering threshold in packets to accelerate fast retransmit.  Time
+threshold loss detection does not require any special ACK generation to
+accelerate fast retransmit, because it uses a timer.
```

Comment:
I think this text should be purely prescriptive, and only from the receiver's point of view, since the sender's is basically not known at the receiver. This also does not say when a receiver should stop immediate acking, which is important IMO.

Suggested replacement for text from "The receiver MAY" : "A receiver SHOULD immediately ack at least three subsequent packets after first receiving a packet out of order, after which it SHOULD resume delaying acknowledgements."